web/src/topic_list_data.ts

```diff
@@ -169,6 +170,11 @@ export function get_list_info(
     const stream_muted = sub.is_muted;
 
     let topic_names = stream_topic_history.get_recent_topic_names(stream_id);
+
+    if (narrowed_topic && !topic_names.includes(narrowed_topic)) {
+        topic_names.unshift(narrowed_topic);
+    }
```

Comment:
What enforces that the current view is also narrowed to the current stream in this logic?

I realize that currently, there may be a global constraint that you can only open a topic while you're viewing it, but let's not hardcode that here in a weird way; check if the `narrow_state.stream`  matches too before doing this `unshift`.

Reply:
I have added a check to ensure that the `stream_id` for topic list matches to that of the narrowed view before adding the topic to `topic_names`.